### ISSUE TITLE
add "use client" to HydrationProvider

### ISSUE THREAD

**Issue by cassus** [traviswimer/react-hydration-provider#6]
Hi @traviswimer,

Thanks for this great tool!

I'm using React 18 with server components in NextJS. I get the following error
```
Server Error
Error: createContext only works in Client Components. Add the "use client" directive at the top of the file to use it. Read more: https://nextjs.org/docs/messages/context-in-server-component
```

Would you add "use client" to `HydrationProvider`?

**Comment by traviswimer** [comment]
Yes, although I think `use client` will be necessary in several other files as well, since `useEffect()` is fundamental to how it works.

I think the library in general should work fine with server components once that's added, although there may be something I'm overlooking.

You can make a PR if you'd like, otherwise I'll try to get this added soon.

**Comment by traviswimer** [comment]
@cassus Just published this in v2.1.0 if you want to try it out.

Also added a small section to the README: https://github.com/traviswimer/react-hydration-provider/tree/v2.1.0#usage-with-server-components

**Comment by cassus** [comment]
Thank you @traviswimer, it works great!
